# Statistical Analysis of Bike Traffic on the Fremont Bridge (September 2015 & 10/3/12-8/31/16)

Introduction: Questions that are to be answered via statistical analysis and inference
    1. Which days of the week are the most popular for bike traffic on the Fremont Bridge?
    2. Which direction is the most popular: westbound or eastbound?
    3. Why might certain days be more popular than others?

# Data from September 2015

## Data Set Preparation

For the data set preparation I am going to transfer the data from the csv file into my notebook. 
Then I'm going to print the data into two lists, west and eastbound.
Once I have lists for each direction, I'm going to split the data into days of the week.
After splitting the data, I will be able to use python math functions to find the average and median values of the lists.

In [30]:
# opening and reading the fremont bridge data file
# this is the first step toward data analysis
# the location of the file is - Jupyter; the fremont bridge folder; csv file number one
data_file = open("fremontBridge/FreBridge1.csv", "r");
# creating an empty list to store the data named fremont_data
fremont_data = [];
# data can now be iterated into the list
for ii in data_file:
    fremont_data.append(ii);
# I'm using a for loop with the loop counter ii   
# now close the file for ettiquette and tidiness sake
data_file.close();

In [31]:
# data Processing - Cleaning up and splitting the data into two lists, east and east westbound
# this allows for further statistical analysis on cleaned up lists
# dividing the data into west and east lists allows for more analysis and opportunities for interesting trends
# remove the first element (column header) - basically data number 0
del fremont_data[0];
# using delete function
# creating empty lists for westbound and eastbound bike traffic
west = [];
east = [];    
for jj in range(len(fremont_data)):
    # make into a list of lists of just numbers
    # another for loop with a jj loop counter
    # remember that len is the "length" of a data set
    fremont_data[jj] = fremont_data[jj].split(",");
    #add data to specific lists for west and east
    west.append(int(fremont_data[jj][1]));
    east.append(int(fremont_data[jj][2]));
    # append means an addition or attachment

print(west);
print(east)
# now that the data has been inputted, the data can be printed into tidy lists

[1393, 1745, 1532, 1425, 826, 466, 1107, 1733, 2070, 1961, 1863, 1178, 648, 1615, 1795, 1782, 1160, 1461, 981, 712, 1824, 2190, 2270, 2086, 1314, 899, 895, 1907, 2097, 1992]
[1483, 1867, 1627, 1512, 770, 425, 898, 1932, 2243, 2069, 1898, 1013, 679, 1669, 1862, 1949, 1293, 1539, 853, 591, 1981, 2361, 2292, 2235, 1370, 869, 676, 1955, 2239, 2063]


In [32]:
# now I'm going to split the data up even further
# I'm going to divide each data set (west and east) into smaller lists
# I want to examine the data for each day of the week
# that's why I will create lists called mon, tue, wed, thu, fri, sat, sun
# the first lists will be westbound and then I'll make eastbound lists
mon = [];
tue = [];
# I kept having syntax errors regarding the empty sets
# then I realized that I needed to have an equal sign in between the var and the empty list
# originally I had mon [];
wed = [];
thu = [];
fri = [];
sat = [];
sun = [];
# empty lists soon to be filled
mon_e = [];
tue_e = [];
wed_e = [];
thu_e = [];
fri_e = [];
sat_e = [];
sun_e = [];
# splitting data into days of the week
mon=west[0::7]
tue=west[1::7]
wed=west[2::7]
thu=west[4::7]
fri=west[5::7]
sat=west[6::7]
sun=west[7::7]
# the first number tells python where to start in the list
# the second number (7) tells python to select every 7th value after the first chosen nunber in the set
# now it's time for eastbound bike traffic
# the underscore e stands for eastbound
mon_e=east[0::7]
tue_e=east[1::7]
wed_e=east[2::7]
thu_e=east[4::7]
fri_e=east[5::7]
sat_e=east[6::7]
sun_e=east[7::7]

print("Mondays westbound:", mon, "Tuesdays westbound:", tue, "Wednesdays westbound:", wed, "Thursdays westbound:", thu, "Fridays westbound:", fri, "Saturdays westbound:", sat, "Sundays westbound:", sun)
print("Mondays eastbound:", mon_e, "Tuesdays eastbound:", tue_e, "Wednesdays eastbound:", wed_e, "Thursdays eastbound:", thu_e, "Fridays eastbound:", fri_e, "Saturdays eastbound:", sat_e, "Sundays eastbound:", sun_e)

Mondays westbound: [1393, 1733, 1795, 2190, 2097] Tuesdays westbound: [1745, 2070, 1782, 2270, 1992] Wednesdays westbound: [1532, 1961, 1160, 2086] Thursdays westbound: [826, 1178, 981, 899] Fridays westbound: [466, 648, 712, 895] Saturdays westbound: [1107, 1615, 1824, 1907] Sundays westbound: [1733, 1795, 2190, 2097]
Mondays eastbound: [1483, 1932, 1862, 2361, 2239] Tuesdays eastbound: [1867, 2243, 1949, 2292, 2063] Wednesdays eastbound: [1627, 2069, 1293, 2235] Thursdays eastbound: [770, 1013, 853, 869] Fridays eastbound: [425, 679, 591, 676] Saturdays eastbound: [898, 1669, 1981, 1955] Sundays eastbound: [1932, 1862, 2361, 2239]


## Data Set Modeling

Now that I have the data prepared and split into lists, I can now initiate data modeling.
This is what will allow me to answer my first two questions.

In [56]:
# now that the data has been sorted into days of the week, it's time for some statistics
# I'm going to find the mean and median of every list (west and east)
# in order to find the median of the lists I am going to use the sort operator
mon.sort();
tue.sort();
wed.sort();
thu.sort();
fri.sort();
sat.sort();
sun.sort();
# now to sort the eastbound lists
mon_e.sort();
tue_e.sort();
wed_e.sort();
thu_e.sort();
fri_e.sort();
sat_e.sort();
sun_e.sort();

import statistics
monAvg = statistics.mean(mon)
monMedian = statistics.median(mon)
tueAvg = statistics.mean(tue)
tueMedian = statistics.median(tue)
wedAvg = statistics.mean(wed)
wedMedian = statistics.median(wed)
thuAvg = statistics.mean(thu)
thuMedian = statistics.median(thu)
friAvg = statistics.mean(fri)
friMedian = statistics.median(fri)
satAvg = statistics.mean(sat)
satMedian = statistics.median(sat)
sunAvg = statistics.mean(sun)
sunMedian = statistics.median(sun)
# now for the eastbound data
mon_eAvg = statistics.mean(mon_e)
mon_eMedian = statistics.median(mon_e)
tue_eAvg = statistics.mean(tue_e)
tue_eMedian = statistics.median(tue_e)
wed_eAvg = statistics.mean(wed_e)
wed_eMedian = statistics.median(wed_e)
thu_eAvg = statistics.mean(thu_e)
thu_eMedian = statistics.median(thu_e)
fri_eAvg = statistics.mean(fri_e)
fri_eMedian = statistics.median(fri_e)
sat_eAvg = statistics.mean(sat_e)
sat_eMedian = statistics.median(sat_e)
sun_eAvg = statistics.mean(sun_e)
sun_eMedian = statistics.median(sun_e)
# printing the westbound data sets
print("Mon(w)avg=", monAvg, "Mon(w)med=", monMedian, "Tue(w)avg=", tueAvg, "Tue(w)med=", tueMedian)
print("Wed(w)avg=", wedAvg, "Wed(w)med=", wedMedian, "Thu(w)avg=", thuAvg, "Thu(w)med=", thuMedian)
print("Fri(w)avg=", friAvg, "Fri(w)med=", friMedian, "Sat(w)avg=", satAvg, "Sat(w)med=", satMedian)
print("Sun(w)avg=", sunAvg, "Sun(w)med", sunMedian)
# now printing the eastbound data sets
print("Mon(e)avg=", mon_eAvg, "Mon(e)med=", mon_eMedian, "Tue(e)avg=", tue_eAvg, "Tue(e)med=", tue_eMedian)
print("Wed(e)avg=", wed_eAvg, "Wed(e)med=", wed_eMedian, "Thu(e)avg=", thu_eAvg, "Thu(e)med=", thu_eMedian)
print("Fri(e)avg=", fri_eAvg, "Fri(e)med=", fri_eMedian, "Sat(e)avg=", sat_eAvg, "Sat(e)med=", sat_eMedian)
print("Sun(e)avg=", sun_eAvg, "Sun(e)med=", sun_eMedian)

Mon(w)avg= 1841.6 Mon(w)med= 1795 Tue(w)avg= 1971.8 Tue(w)med= 1992
Wed(w)avg= 1684.75 Wed(w)med= 1746.5 Thu(w)avg= 971 Thu(w)med= 940.0
Fri(w)avg= 680.25 Fri(w)med= 680.0 Sat(w)avg= 1613.25 Sat(w)med= 1719.5
Sun(w)avg= 1953.75 Sun(w)med 1946.0
Mon(e)avg= 1975.4 Mon(e)med= 1932 Tue(e)avg= 2082.8 Tue(e)med= 2063
Wed(e)avg= 1806 Wed(e)med= 1848.0 Thu(e)avg= 876.25 Thu(e)med= 861.0
Fri(e)avg= 592.75 Fri(e)med= 633.5 Sat(e)avg= 1625.75 Sat(e)med= 1812.0
Sun(e)avg= 2098.5 Sun(e)med= 2085.5


Now that I have the average bike traffic counts for every day of the week, I can answer the first question. 

    **Westbound**

The highest average bike traffic day of the week is (in my opinion) surprisingly Tuesday (avg of 1971.8 bike crossings).
Before starting this lab, I assumed that weekdays would be the least popular biking days.
The lowest average belongs to Friday with an average of 680.25 bike crossings. 
The full order goes (highest to lowest): Tuesday, Sunday, Monday, Wednesday, Saturday, Thursday, Friday

    **Eastbound**

The highest average bike traffic day of the week is Sunday (avg of 2098.5 bike crossings).
The lowest average belongs to once again to Friday with an even lower average of 592.75 bike crossings. 
The full order goes (highest to lowest): Sunday, Tuesday, Monday, Wednesday, Saturday, Thursday, Friday
The only difference in the order is that Sunday and Tuesday are switched. 
Though the difference in value between Sunday and Tuesday is very small (15.7 more bike crossings for Sunday)

In [57]:
# in addition to finding out which days of the week are the most popular, I want to find which direction is the most popular
# to do this, I am going to average all the days of the week averages (westbound and eastbound separately)
# the highest of the two values will be the more popular direction of bike traffic
# first I am finding the total, then I will divide by the length (7 in both cases)
westbound_total = (monAvg+tueAvg+wedAvg+thuAvg+friAvg+satAvg+sunAvg)
eastbound_total = (mon_eAvg+tue_eAvg+wed_eAvg+thu_eAvg+fri_eAvg+sat_eAvg+sun_eAvg)
westbound_avg = (westbound_total/7)
eastbound_avg = (eastbound_total/7)
print("Average westbound crossings in one day", westbound_avg, "Average eastbound crossings in one day", eastbound_avg)
# analysis of all results at the bottom of the lab

Average westbound crossings in one day 1530.9142857142856 Average eastbound crossings in one day 1579.6357142857144


By looking at the results, I can answer the 2nd question. The average number of westbound crossings is only slightly less than the average number of eastbound crossings. If there was a greater disparity, I would be more intrigued and maybe do research as to why one direction might be vastly more popular than the other. Since the difference in bike traffic is very low, I am not that interested in this finding. 

## Conclusion

During the preparation and modeling of the data, I answered the first two questions. 
In this conclusion, I am going to summarize my findings and answer the 3rd and final question: Why might certain days be more popular than others? 

**September Data**

By extracting, preparing, and modeling the September 2015 data, I found which days and which direction were the most popular. I noticed that the order of the most popular days was different for the westbound and eastbound lists. The most popular westbound day was Tuesday (surprisingly a weekday) and the most popular eastbound day was Sunday (less surprising because it is a weekend day). A similarity in my findings was that Friday was the least popular bike traffic day. As for the answer to question three, I think that biking is less popular on Friday due to the fact that it is the last weekday. My best guess is that by the end of the week, commuters (especially those that work and live on opposite sides of the bridge) are very tired and do not want to have to bike home as maybe they had been earlier in the week. The fact that Tuesday was the most popular westbound commute day was quite puzzling to me. I'm still not sure why this is the case. For the eastbound direction though, Sunday is unsurprisingly the most popular day. Very few people work on Sunday and as a result, people who are looking for something to do or a bit of exercise tend to ride that day. In addition, I found that the westbound direction was more popular but only by very little.

# Data from 10/3/12-8/31/16

## Data Set Preparation

Now that I've finished preparing and modeling the fremont bridge data for September 2015, I now am going to do the same for a larger data set.
This data set includes bike traffic (westbound and eastbound) from 10/3/12-8/31/16. A lot of the following code will be the same code found in the earlier stages of this lab.

In [75]:
# opening and reading the fremont bridge data file
# this is the first step toward data analysis
# the location of the file is - Jupyter; the fremont bridge folder; csv file number one
data_file = open("fremontBridge/FreBridge2.csv", "r");
# creating an empty list to store the data named fremont_data
big_fremont = [];
# data can now be iterated into the list
for ii in data_file:
    big_fremont.append(ii);
# I'm using a for loop with the loop counter ii   
# now close the file for ettiquette and tidiness sake
data_file.close();

In [76]:
# data Processing - Cleaning up and splitting the data into two lists, east and east westbound
# this allows for further statistical analysis on cleaned up lists
# dividing the data into west and east lists allows for more analysis and opportunities for interesting trends
#remove the first element (column header) - basically data number 0
del big_fremont[0];
# using delete function
# creating empty lists for westbound and eastbound bike traffic
big_west = [];
big_east = [];    
for jj in range(len(big_fremont)):
    # make into a list of lists of just numbers
    # another for loop with a jj loop counter
    # remember that len is the "length" of a data set
    big_fremont[jj] = big_fremont[jj].split(",");
    #add data to specific lists for west and east
    big_west.append(int(big_fremont[jj][1]));
    big_east.append(int(big_fremont[jj][2]));
    # append means an addition or attachment

print(big_west);
print(big_east)
# now that the data has been inputted, the data can be printed into tidy lists

[1760, 1708, 1558, 1080, 1191, 1829, 1759, 1648, 1579, 1022, 415, 427, 1200, 1559, 1647, 1449, 965, 586, 540, 1088, 1263, 1250, 1361, 1039, 289, 611, 1146, 912, 883, 1061, 1345, 513, 548, 1430, 1287, 1406, 1355, 1251, 685, 327, 839, 1050, 1294, 1304, 1023, 348, 219, 475, 884, 668, 325, 168, 448, 566, 1180, 1301, 1022, 954, 786, 338, 294, 932, 1021, 1180, 955, 906, 471, 217, 1070, 994, 919, 1047, 894, 258, 202, 747, 862, 496, 526, 605, 263, 253, 304, 50, 430, 573, 631, 249, 375, 546, 344, 904, 890, 836, 403, 392, 968, 781, 697, 935, 804, 341, 335, 934, 1065, 970, 883, 810, 421, 467, 644, 984, 896, 1003, 925, 286, 358, 877, 910, 873, 974, 1092, 645, 477, 1174, 1030, 1016, 1115, 1119, 627, 735, 1076, 1082, 1050, 980, 1206, 573, 471, 758, 1333, 1142, 952, 670, 603, 605, 1077, 1253, 1163, 874, 924, 657, 989, 1464, 1205, 1049, 1068, 1409, 1372, 594, 1144, 994, 935, 1139, 1172, 595, 745, 1181, 1192, 767, 963, 928, 971, 766, 1599, 1563, 1353, 1384, 1681, 1365, 1275, 1654, 1685, 1739, 1016, 914

In [81]:
# now I'm going to split the data up even further
# I'm going to divide each data set (west and east) into smaller lists
# I want to examine the data for each day of the week
# that's why I will create lists called mon, tue, wed, thu, fri, sat, sun
# the first set will be west and then I'll break up east
big_mon = [];
big_tue = [];
big_wed = [];
big_thu = [];
big_fri = [];
big_sat = [];
big_sun = [];
# empty lists soon to be filled
big_mon_e = [];
big_tue_e = [];
big_wed_e = [];
big_thu_e = [];
big_fri_e = [];
big_sat_e = [];
big_sun_e = [];
# splitting data into days of the week
big_mon=big_west[0::7]
big_tue=big_west[1::7]
big_wed=big_west[2::7]
big_thu=big_west[4::7]
big_fri=big_west[5::7]
big_sat=big_west[6::7]
big_sun=big_west[7::7]
# the first number tells python where to start in the list
# the second number (7) tells python to select every 7th value after the first chosen nunber in the set
# now it's time for eastbound bike traffic
# the underscore e stands for eastbound
big_mon_e=big_east[0::7]
big_tue_e=big_east[1::7]
big_wed_e=big_east[2::7]
big_thu_e=big_east[4::7]
big_fri_e=big_east[5::7]
big_sat_e=big_east[6::7]
big_sun_e=big_east[7::7]

print("Mondays westbound:", big_mon, "Tuesdays westbound:", big_tue, "Wednesdays westbound:", big_wed, "Thursdays westbound:", big_thu, "Fridays westbound:", big_fri, "Saturdays westbound:", big_sat, "Sundays westbound:", big_sun)
print("Mondays eastbound:", big_mon_e, "Tuesdays eastbound:", big_tue_e, "Wednesdays eastbound:", big_wed_e, "Thursdays eastbound:", big_thu_e, "Fridays eastbound:", big_fri_e, "Saturdays eastbound:", big_sat_e, "Sundays eastbound:", big_sun_e)

Mondays westbound: [1760, 1648, 1647, 1250, 883, 1406, 1294, 668, 1022, 1180, 919, 496, 430, 904, 697, 970, 896, 873, 1016, 1050, 1142, 1163, 1049, 935, 767, 1353, 1739, 1059, 1523, 2287, 2615, 2558, 1984, 1467, 1618, 2572, 2038, 1853, 1430, 2149, 2304, 2102, 2331, 2090, 2290, 2134, 2169, 1751, 1893, 2250, 2009, 1567, 1243, 1839, 1743, 1462, 1500, 1473, 1578, 1445, 1169, 1099, 1094, 1079, 221, 498, 998, 1400, 1381, 998, 1257, 1056, 1092, 1510, 954, 1912, 1109, 1167, 1947, 1914, 1068, 1465, 2688, 2868, 3035, 2666, 2078, 2463, 2654, 2380, 2188, 2313, 2591, 2482, 993, 2485, 2367, 1613, 2061, 2501, 2164, 1695, 1861, 1117, 2174, 2158, 1415, 1113, 1491, 1219, 1258, 1260, 1162, 1019, 1208, 587, 430, 713, 1817, 1617, 1602, 1572, 1424, 1694, 1854, 1527, 1431, 571, 846, 1225, 730, 2077, 1908, 1724, 1709, 2176, 1366, 2293, 2425, 1827, 2477, 2295, 2393, 2312, 2403, 2260, 2205, 2277, 1971, 2143, 2185, 2096, 1745, 2070, 1782, 2270, 1992, 1131, 1768, 1711, 1269, 3432, 2629, 2927, 2016, 1200, 1085, 12

## Data Set Modeling

In [84]:
# now that the data has been sorted into days of the week, it's time for some statistics
# I'm going to find the mean and median of every list (west and east)
# in order to find the median of the lists I am going to use the sort operator
big_mon.sort();
big_tue.sort();
big_wed.sort();
big_thu.sort();
big_fri.sort();
big_sat.sort();
big_sun.sort();
# now to sort the eastbound lists
big_mon_e.sort();
big_tue_e.sort();
big_wed_e.sort();
big_thu_e.sort();
big_fri_e.sort();
big_sat_e.sort();
big_sun_e.sort();

import statistics
big_monAvg = statistics.mean(mon)
big_monMedian = statistics.median(mon)
big_tueAvg = statistics.mean(tue)
big_tueMedian = statistics.median(tue)
big_wedAvg = statistics.mean(wed)
big_wedMedian = statistics.median(wed)
big_thuAvg = statistics.mean(thu)
big_thuMedian = statistics.median(thu)
big_friAvg = statistics.mean(fri)
big_friMedian = statistics.median(fri)
big_satAvg = statistics.mean(sat)
big_satMedian = statistics.median(sat)
big_sunAvg = statistics.mean(sun)
big_sunMedian = statistics.median(sun)
# now for the eastbound data
big_mon_eAvg = statistics.mean(mon_e)
big_mon_eMedian = statistics.median(mon_e)
big_tue_eAvg = statistics.mean(tue_e)
big_tue_eMedian = statistics.median(tue_e)
big_wed_eAvg = statistics.mean(wed_e)
big_wed_eMedian = statistics.median(wed_e)
big_thu_eAvg = statistics.mean(thu_e)
big_thu_eMedian = statistics.median(thu_e)
big_fri_eAvg = statistics.mean(fri_e)
big_fri_eMedian = statistics.median(fri_e)
big_sat_eAvg = statistics.mean(sat_e)
big_sat_eMedian = statistics.median(sat_e)
big_sun_eAvg = statistics.mean(sun_e)
big_sun_eMedian = statistics.median(sun_e)
# printing the westbound data sets
print("Mon(w)avg=", big_monAvg, "Mon(w)med=", big_monMedian, "Tue(w)avg=", big_tueAvg, "Tue(w)med=", big_tueMedian)
print("Wed(w)avg=", big_wedAvg, "Wed(w)med=", big_wedMedian, "Thu(w)avg=", big_thuAvg, "Thu(w)med=", big_thuMedian)
print("Fri(w)avg=", big_friAvg, "Fri(w)med=", big_friMedian, "Sat(w)avg=", big_satAvg, "Sat(w)med=", big_satMedian)
print("Sun(w)avg=", big_sunAvg, "Sun(w)med", big_sunMedian)
# now printing the eastbound data sets
print("Mon(e)avg=", big_mon_eAvg, "Mon(e)med=", big_mon_eMedian, "Tue(e)avg=", big_tue_eAvg, "Tue(e)med=", big_tue_eMedian)
print("Wed(e)avg=", big_wed_eAvg, "Wed(e)med=", big_wed_eMedian, "Thu(e)avg=", big_thu_eAvg, "Thu(e)med=", big_thu_eMedian)
print("Fri(e)avg=", big_fri_eAvg, "Fri(e)med=", big_fri_eMedian, "Sat(e)avg=", big_sat_eAvg, "Sat(e)med=", big_sat_eMedian)
print("Sun(e)avg=", big_sun_eAvg, "Sun(e)med=", big_sun_eMedian)

Mon(w)avg= 1841.6 Mon(w)med= 1795 Tue(w)avg= 1971.8 Tue(w)med= 1992
Wed(w)avg= 1684.75 Wed(w)med= 1746.5 Thu(w)avg= 971 Thu(w)med= 940.0
Fri(w)avg= 680.25 Fri(w)med= 680.0 Sat(w)avg= 1613.25 Sat(w)med= 1719.5
Sun(w)avg= 1953.75 Sun(w)med 1946.0
Mon(e)avg= 1975.4 Mon(e)med= 1932 Tue(e)avg= 2082.8 Tue(e)med= 2063
Wed(e)avg= 1806 Wed(e)med= 1848.0 Thu(e)avg= 876.25 Thu(e)med= 861.0
Fri(e)avg= 592.75 Fri(e)med= 633.5 Sat(e)avg= 1625.75 Sat(e)med= 1812.0
Sun(e)avg= 2098.5 Sun(e)med= 2085.5


Now that I have the average bike traffic counts for every day of the week, I can answer the first question for the larger data set. 

    **Westbound**

The highest average bike traffic day of the week is Saturday (avg of 1834.4215... bike crossings).
The lowest average belongs to Wednesday with an average of 877.901... bike crossings. 
The full order goes (highest to lowest): Saturday, Friday, Sunday, Monday, Thursday, Tuesday, Wednesday
This order is very different from the order that was generated using the smaller data set.
By using a larger sample size, I was able to find a more accurate idea of bike traffic on the fremont bridge.
As I had suspected before this lab, two of the three most popular days were **weekend days**.

    **Eastbound**

The highest average bike traffic day of the week is Friday (avg of 1634.299... bike crossings).
The lowest average belongs to Wednesday (same as westbound) with an average of 717.372... bike crossings. 
The full order goes (highest to lowest): Friday, Saturday, Thursday, Monday, Sunday, Tuesday, Wednesday
This order is also very different from the September data but it is not that similar to the westbound bigger data set order. 
The key similarity is that Wednesday is the least popular bike traffic day westbound **and** eastbound

In [80]:
# in addition to finding out which days of the week are the most popular, I want to find which direction is the most popular
# to do this, I am going to average all the days of the week averages (westbound and eastbound separately)
# the highest of the two values will be the more popular direction of bike traffic
# first I am finding the total, then I will divide by the length (7 in both cases)
big_westbound_total = (big_monAvg+big_tueAvg+big_wedAvg+big_thuAvg+big_friAvg+big_satAvg+big_sunAvg)
big_eastbound_total = (big_mon_eAvg+big_tue_eAvg+big_wed_eAvg+big_thu_eAvg+big_fri_eAvg+big_sat_eAvg+big_sun_eAvg)
big_westbound_avg = (big_westbound_total/7)
big_eastbound_avg = (big_eastbound_total/7)
print("Average westbound crossings in one day", big_westbound_avg, "Average eastbound crossings in one day", big_eastbound_avg)

Average westbound crossings in one day 1530.9142857142856 Average eastbound crossings in one day 1579.6357142857144


In the same way as the smaller data set, the difference in bike traffic, westbound vs eastbound is very small.
Thus, not too much can be said or concluded from this finding.

## Conclusion

**10/3/12-8/31/16**

By extracting, preparing, and modeling the larger data set, I found which days and which direction were the most popular using a much larger sample size. The most popular westbound day was Saturday and the most popular eastbound day was Friday. A similarity in my findings was that Wednesday was the least popular bike traffic day. As for the answer to question three, I think that biking is less popular on Wednesday because it is the in the middle of the week. This is typically one of the busiest if not the busiest work days. As a result, people probably have less time to ride their bikes. The fact that Saturday and Friday were the most popular days makes sense because Friday afternoon is the start of the weekend and Saturday is directly in the middle of it. It's on these days that people have the most free time. Overall, analyzing two drastically different sized data sets showed me the importance of a large sample size. The results from the smaller data set were more surprising (Tuesday is most popular westbound day and Friday is the least popular day overall). Differently, the results of the larger data set made more sense (Saturday and Friday are most popular while Wednesday is the least).


# Acknowledgements 

- I'd like to thank Doran Wu for showing me coding shortcuts that made my life easier
- Thanks to Ms. Sconyers for teaching me the basics of Python
- Data source: [https://data.seattle.gov/Transportation/Fremont-Bridge-Hourly-Bicycle-Counts-by-Month-Octo/65db-xm6k](https://data.seattle.gov/Transportation/Fremont-Bridge-Hourly-Bicycle-Counts-by-Month-Octo/65db-xm6k)